### ISSUE TITLE
Hive files exposed to user on (theoricaly all) platforms

### ISSUE THREAD

**Issue by QuentinCG** [hivedb/hive#1044]
**Steps to Reproduce**
Do `Hive.initFlutter()`

**Code sample**

Hive.initFlutter() is using `getApplicationDocumentsDirectory` which is designed to provide a folder accessible to users (source: "path_provider" package) !
(go to `More info` section to see the folders depending on the platform)

**Version**
 - Platform: Mac, Windows, Linux, (iOS and Android theoricaly but not in practice even though it seems to impact some people: https://github.com/hivedb/hive/issues/840)
 - Flutter version: [3.0.5]
 - Hive version: [all, including 2.2.3]

**How to fix**

Use `getApplicationSupportDirectory` instead of `getApplicationDocumentsDirectory` at https://github.com/hivedb/hive/blob/3b12c31a221f97f5ec86fe20df63515aeedf88f0/hive_flutter/lib/src/hive_extensions.dart#L16:

**Ressources**

path_provider documentation:

```
/// Use this for files you don’t want exposed to the user. Your app should not
/// use this directory for user data files.
/// (...)
getApplicationSupportDirectory
```

```
/// Path to a directory where the application may place data that is
/// user-generated, or that cannot otherwise be recreated by your application.
/// On iOS, this uses the `NSDocumentDirectory` API. Consider using
/// [getApplicationSupportDirectory] instead if the data is not user-generated.
/// (...)
getApplicationDocumentsDirectory
```

**More Info**

Here are where data are stored depending on the function called:

getApplicationDocumentsDirectory:
- Windows: "C/Users/{username}/Documents/" (User accessible 👎 )
- Linux: "/home/{user}/"  (User accessible 👎 )
- Android : "/data/user/0/{YourAppName}/app_flutter/" (Not really accessible so I guess it is "ok" 👍)
- MacOS (Not tested, but I guess, it is like Linux & Windows 👎 )
- IOS (Not tested, but from documentation, it is specified we should NOT use this function... 👎 )

getApplicationSupportDirectory :
- Android: "/data/user/0/{YourAppName}/files/" (Not accessible 👍)
- "C:\Users\{user}\AppData\Roaming\{YourAppOrganization}\{YourAppName}/" (Not accessible 👍 )
- "/home/{user}/.local/share/{YourAppName]/app_database.db" pour Linux  (Not accessible 👍 )
- MacOS (Not tested, but I guess, it is like Linux & Windows 👍 )
- IOS (Not tested, but from documentation, it is specified we SHOULD use this function... 👍 )

**Comment by QuentinCG** [comment]
Other consequence on Windows/Linux/MacOS: If 2 apps uses the same "box name", they will share it...

**Comment by manafire** [comment]
Agreed.  There's been some ongoing discussion on it for some time now: https://github.com/hivedb/hive/pull/746.
There's a comment in there that you can apparently use Hive.init and pass it whatever directory you'd like since `initFlutter` effectively calls `init` anyway.  e.g.:

```dart
  final suppDir = await getApplicationSupportDirectory();
  Hive.init(suppDir.path);
```

I've never used Hive before so hopefully it works well; migration would be another case.

**Comment by PhilipPurwoko** [comment]
In Flutter I cloud set the path so it was not visible in Files
```dart
String? path = (await getApplicationSupportDirectory()).path;
await Hive.initFlutter();
await Hive.openBox(boxApp, path: path);
await Hive.openBox(boxUser, path: path);
```
